lua/frecency/util.lua

```diff
@@ -0,0 +1,111 @@
+local uv = vim.loop
+local const = require"frecency.const"
+local Path = require"plenary.path"
+
+local util = {}
+
+-- stolen from penlight
+
+---escape any Lua 'magic' characters in a string
+util.escape = function(str)
+  return (str:gsub("[%-%.%+%[%]%(%)%$%^%%%?%*]", "%%%1"))
+end
+
+util.string_isempty = function(str)
+  return str == nil or str == ""
+end
+
+util.filemask = function(mask)
+  mask = util.escape(mask)
+  return "^" .. mask:gsub("%%%*", ".*"):gsub("%%%?", ".") .. "$"
+end
+
+util.file_is_ignored = function(filepath, ignore_patters)
+  local is_ignored = false
+  --- TODO: it should be merged or flatten
+  for _, pattern in pairs(ignore_patterns and ignore_patters or const.ignore_patterns) do
+    if filepath:find(util.filemask(pattern)) ~= nil then
+      is_ignored = true
+      goto continue
+    end
+  end
+
+  ::continue::
+  return is_ignored
+end
+
+util.fs_stat = function(path)
+  local stat = path and uv.fs_stat(path) or nil
+  local res = {}
+  res.exists = stat and true or false -- TODO: this is silly
+  res.isdirectory = (stat and stat.type == "directory") and true or false
+
+  return res
+end
+
+util.path_invalid = function(path)
+  local stat = util.fs_stat(path)
+  if stat == {} or stat.isdirectory then
+    return true
+  end
+  return false
+end
+
+util.confirm_deletion = function (num_of_entries)
+  local question = "Telescope-Frecency: remove %d entries from SQLite3 database?"
+  return vim.fn.confirm(question:format(num_of_entries), "&Yes\n&No", 2) == 1
+end
+
+util.abort_remove_unlinked_files = function()
+  ---TODO: refactor all messages to a lua file. alarts.lua?
+  print "TelescopeFrecency: validation aborted."
+end
+
+util.tbl_match = function(field, val, tbl)
+  return vim.tbl_filter(function(t)
+    return t[field] == val
+  end, tbl)
+end
+
+---Wrappe around Path:new():make_relative
+---@return string
+---@FIXME: errors out, path is nil
+util.path_make_relative = function (path, cwd)
+  return Path:new(cwd..path):make_relative()
```

Comment:
this is a little bit better. `Path:new({ cwd, path })`

Reply:
Interesting I didn't know that was possible.